src/components/interaction-designer/block-editors/GroupMembershipEditor.vue

```diff
@@ -1,153 +1,204 @@
-            :value="MEMBERSHIP_ACTION.SET_FROM_EXPRESSION"
-            class="custom-control-input">
-          <label
-            class="custom-control-label font-weight-normal"
-            for="setFromExpression">
-            {{ 'flow-builder.set-group-membership-from-expression' | trans }}
+            :for="action.id">
+            {{ action.label }}
           </label>
         </div>
       </div>
 
-      <validation-message
-        #input-control="{ isValid }"
-        :message-key="`block/${block.uuid}/config/group_key`">
-        <div class="block-group-key">
-          <text-editor
-            v-model="groupKey"
-            :label="'flow-builder.group-label' | trans"
-            :label-class="'font-weight-bold'"
-            :placeholder="'flow-builder.enter-group-label' | trans"
-            :valid-state="isValid" />
-        </div>
-      </validation-message>
-
-      <validation-message
-        v-if="membershipAction === MEMBERSHIP_ACTION.SET_FROM_EXPRESSION"
-        #input-control="{ isValid }"
-        :message-key="`block/${block.uuid}/config/is_member`">
-        <expression-input
-          :label="'flow-builder.value-expression' | trans"
-          :placeholder="'flow-builder.enter-expression' | trans"
-          :label-class="'font-weight-bold'"
-          :current-expression="isMember"
-          :valid-state="isValid"
-          @commitExpressionChange="updateIsMemberExpression" />
-      </validation-message>
+      <vue-multiselect
+        v-if="isGroupListVisible"
+        v-model="selectedGroups"
+        :internal-search="!hasCustomSearch"
+        :is-loading="groupsLoading"
+        :options="groupOptions"
+        :taggable="!hasCustomSearch"
+        :multiple="true"
+        track-by="group_key"
+        label="group_name"
+        @seach-change="onSearchChange"
+        @tag="onGroupAdd" />
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import {IBlock} from '@floip/flow-runner'
+import {IBlock, ISetGroupMembershipBlockConfig, IGroupMembership} from '@floip/flow-runner'
 import {Component, Prop} from 'vue-property-decorator'
 import {namespace} from 'vuex-class'
+import VueMultiselect from 'vue-multiselect'
 import Lang from '@/lib/filters/lang'
-import {get} from 'lodash'
 import {mixins} from 'vue-class-component'
 
 const flowVuexNamespace = namespace('flow')
 
-const NULL_STRING_EXPRESSION = '@(null)'
-const TRUTHY_STRING_EXPRESSION = '@(true)'
-const EMPTY_STRING_EXPRESSION = ''
+type MembershipAction = {
+  id: string,
+  label: string,
+  value: string,
+}
+
+enum MEMBERSHIP_ACTION {
+  ADD = 'add',
+  REMOVE = 'remove',
+  CLEAR = 'clear',
+}
 
-@Component({})
+@Component({
+  components: {VueMultiselect},
+})
 export class GroupMembershipEditor extends mixins(Lang) {
   @Prop() readonly block!: IBlock
 
-  MEMBERSHIP_ACTION = {
-    ADD: 'add',
-    SET_FROM_EXPRESSION: 'set_from_expression',
-    REMOVE: 'remove',
+  /*
+   * The following two props should be used together
```

Comment:
I see want you mean here, because:
- community flow-builder uses a free list
- but voto5 will use a limited list

Thanks for adding these comments, very helpful. Could I ask you an extra favour? Pls add a TODO here
> TODO in VMO-6299: Document these 02 props in stories

In fact, I created this ticket [vmo-6299](https://viamoinc.atlassian.net/browse/VMO-6299) to help update existing stories

- [ ] add a TODO